app/models/comment.php

```diff
@@ -20,17 +20,111 @@ public static function getComments($offset, $limit, $id)
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
         foreach ($rows as $row) {
-           $comments[] = new Comment($row);
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));     
+            $total_favorites = Comment::countFavorites($row['id']);     
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];  
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
         }
         return $comments;
     }
 
-    public static function countComments($thread_id)
+    public static function totalComments($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public function editComment($comment_id)
```

Comment:
Regarding the function names, please refer to this: https://github.com/ezrafinones/FAB-Talk/pull/2/files#r27790481
:dart: